File: telegram/src/main/java/by/mrrockka/service/help/HelpTelegramService.java
package by.mrrockka.service.help;

import by.mrrockka.domain.MessageMetadata;
import by.mrrockka.parser.HelpMessageParser;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.stereotype.Service;
import org.telegram.telegrambots.meta.api.methods.botapimethods.BotApiMethodMessage;
import org.telegram.telegrambots.meta.api.methods.send.SendMessage;

@Slf4j
@Service
@RequiredArgsConstructor
public class HelpTelegramService {
  private static final String HELP_COMMAND = "help";

  private final BotDescriptionProperties botDescriptionProperties;
  private final HelpMessageParser helpMessageParser;

  public BotApiMethodMessage sendHelpInformation(final MessageMetadata messageMetadata) {
    final var description = helpMessageParser.parse(messageMetadata)
      .map(command -> botDescriptionProperties.getCommands().get(command))
      .orElse(botDescriptionProperties.getCommands().get(HELP_COMMAND));

    return SendMessage.builder()
      .chatId(messageMetadata.getChatId())
      .text(description.details())
      .build();
  }

}

File: telegram/src/main/java/by/mrrockka/service/HelpTelegramService.kt
package by.mrrockka.service

import by.mrrockka.BotCommands
import by.mrrockka.domain.MessageMetadata
import by.mrrockka.parser.HelpMessageParser
import org.springframework.stereotype.Service

interface HelpTelegramService {
    fun help(metadata: MessageMetadata): BotCommands.Description
}

@Service
class HelpTelegramServiceImpl(
        private val descriptions: BotCommands,
        private val helpMessageParser: HelpMessageParser,
) : HelpTelegramService {

    override fun help(metadata: MessageMetadata): BotCommands.Description {
        val description = helpMessageParser.parse(metadata)
                .let { text -> descriptions.byNameAndAlias[text] }

        return description ?: descriptions.byNameAndAlias["help"]!!
    }
}
